src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -170,6 +178,25 @@ private bool CheckOption(Workspace workspace, string language, bool documentOpen
             }
         }
 
+        private void OnDocumentOrProjectAnalyzed(Guid documentOrProjectGuid, Project project)
+        {
+            if (!project.SupportsCompilation)
+            {
+                return;
+            }
+
+            var pendingIds = _pendingDocumentsOrProject.GetValue(project,
+                    p => new HashSet<Guid>(p.Documents.Select(d => d.Id.Id).Concat(p.Id.Id)));
+            pendingIds.Remove(documentOrProjectGuid);
+
+            if (pendingIds.Count == 0)
+            {
+                // PERF: We have computed and cached all documents and project diagnostics for the given project, so drop the CompilationWithAnalyzers instance that also caches all these diagnostics.
```

Comment:
@heejaechang @JohnHamby @srivatsn I have added the `_pendingDocumentsOrProject` map to track all the documents/project diagnostics that are pending to be computed. Once a project has been fully processed by the SolutionCrawler, we explicitly dispose off the CompilationWithAnalyzers instance used to compute project's diagnostics, so as to reduce memory pressure from caching diagnostics in both IDE layer as well as CompilationWithAnalyzers diagnostic cache.